File: 프로그래머스/Lv.2/12953. N개의 최소공배수/N개의 최소공배수.java
import java.util.*;
import java.io.*;

class Solution {
    public int solution(int[] arr) {
        int answer = arr[0];
        for(int i=1; i<arr.length; i++){
            answer = lcm(answer,arr[i]);
        }
        return answer;
    }
    
    static int gcd(int a, int b){
        while(b != 0){
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }
    
    static int lcm(int a, int b){
        return a*b / gcd(a,b);
    }
}

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
function solution(arr) {
    var answer = arr[0];
    
    for(let i=1; i<arr.length;i++){
        answer = lcm(answer,arr[i]);
    }
    return answer;
}

function gcd(a,b){
        
    if(b == 0){
        return a;
    }
    
    return gcd(b, a % b);
}

function lcm(a,b){
    return (a*b)/gcd(a,b);
}